### PR TITLE
Finish creating/showing/deleting a user function

### PR DESCRIPTION
**Pull request by khanhkhanh**

### CODE REVIEW

File: app/controllers/supervisor/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+class Supervisor::UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+  
+  def index
+    @users = User.paginate(page: params[:page])
```
Comment (dieunb) [inline app/controllers/supervisor/users_controller.rb]: remove `()`

File: app/controllers/supervisor/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  end
+
+  def show
+    @user = User.find(params[:id])
```
Comment (dieunb) [inline app/controllers/supervisor/users_controller.rb]: remove `()`

File: app/controllers/supervisor/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  end
+
+  def create
+    @user = User.new(user_params)
```
Comment (dieunb) [inline app/controllers/supervisor/users_controller.rb]: remove `()`

File: app/controllers/supervisor/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    else
+      render 'new'
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if @user.update_attributes(user_params)
```
Comment (dieunb) [inline app/controllers/supervisor/users_controller.rb]: remove `()`

File: app/controllers/supervisor/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      redirect_to supervisor_users_path
+    else
+      render 'new'
+    end
+  end
+
+  def edit
```
Comment (dieunb) [inline app/controllers/supervisor/users_controller.rb]: remove

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.paginate(page: params[:page])
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: remove `()`

File: app/views/shared/_error_messages.html.erb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      The form contains <%= pluralize @user.errors.count, "error" %>.
+    </div>
+    <ul>
+    <% @user.errors.full_messages.each do |msg| %>
```
Comment (dieunb) [inline app/views/shared/_error_messages.html.erb]: indent

File: app/views/supervisor/users/_user.html.erb [hunk 8]
```diff
@@ -0,0 +1,6 @@
+<li>
+  <%= gravatar_for user, size: 50 %>
+  <%= link_to user.name, user %>
+  | <%= link_to "delete", supervisor_user_path(user), method: :delete,
+                                data: { confirm: "Are you sure?" } %>
```
Comment (dieunb) [inline app/views/supervisor/users/_user.html.erb]: coding convention

File: app/views/supervisor/users/edit.html.erb [hunk 9]
```diff
@@ -0,0 +1,29 @@
+<% provide(:title, "Edit user") %>
```
Comment (dieunb) [inline app/views/supervisor/users/edit.html.erb]: remove `()`

File: app/views/supervisor/users/index.html.erb [hunk 10]
```diff
@@ -0,0 +1,10 @@
+<% provide(:title, 'All users') %>
```
Comment (dieunb) [inline app/views/supervisor/users/index.html.erb]: remove `()`

File: app/views/supervisor/users/index.html.erb [hunk 11]
```diff
@@ -0,0 +1,10 @@
...
+<h1>All users</h1>
+
+<%= will_paginate %>
+
+  <ul class="users">
```
Comment (dieunb) [inline app/views/supervisor/users/index.html.erb]: indent

File: app/views/users/index.html.erb [hunk 12]
```diff
@@ -0,0 +1,10 @@
+<% provide(:title, 'All users') %>
```
Comment (dieunb) [inline app/views/users/index.html.erb]: remove `()`

File: app/views/users/index.html.erb [hunk 13]
```diff
@@ -0,0 +1,10 @@
...
+<h1>All users</h1>
+
+<%= will_paginate %>
+
+  <ul class="users">
```
Comment (dieunb) [inline app/views/users/index.html.erb]: indent

File: config/routes.rb [hunk 14]
```diff
@@ -1,4 +1,10 @@
 Rails.application.routes.draw do
   root             'static_pages#home'
   get 'help'    => 'static_pages#help'
+
+  resources :users
+
+  namespace :supervisor do
```
Comment (dieunb) [inline config/routes.rb]: root page of namespace???

File: app/controllers/supervisor/users_controller.rb [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  def show
+    @user = User.find params[:id]
+  end
+
+  def create
+    @user = User.new user_params
+    if @user.save
+      redirect_to supervisor_users_path
```
Comment (dieunb) [inline app/controllers/supervisor/users_controller.rb]: flash message??? remember, using i18n

File: app/controllers/supervisor/users_controller.rb [hunk 16]
```diff
@@ -0,0 +1,34 @@
+class Supervisor::UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+  
+  def index
```
Comment (dieunb) [inline app/controllers/supervisor/users_controller.rb]: order actions is following index, new, show, create, edit, update, destroy

File: config/routes.rb [hunk 17]
```diff
@@ -1,4 +1,11 @@
 Rails.application.routes.draw do
-  root             'static_pages#home'
-  get 'help'    => 'static_pages#help'
+  root 'static_pages#home'
+  get 'help' => 'static_pages#help'
+
+  resources :users
```
Comment (dieunb) [inline config/routes.rb]: filter resource by using `only` or `except`

File: config/routes.rb [hunk 18]
```diff
@@ -1,4 +1,11 @@
 Rails.application.routes.draw do
-  root             'static_pages#home'
-  get 'help'    => 'static_pages#help'
+  root 'static_pages#home'
+  get 'help' => 'static_pages#help'
+
+  resources :users, :only => [:index, :show]
```
Comment (dieunb) [inline config/routes.rb]: coding convention

File: app/controllers/supervisor/users_controller.rb [hunk 19]
```diff
@@ -0,0 +1,35 @@
+class Supervisor::UsersController < ApplicationController
+  def index
+    @users = User.paginate page: params[:page]
```
Comment (dieunb) [inline app/controllers/supervisor/users_controller.rb]: per_page???

File: app/controllers/users_controller.rb [hunk 20]
```diff
@@ -0,0 +1,9 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.paginate page: params[:page]
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: per_page???

File: app/views/supervisor/users/new.html.erb [hunk 21]
```diff
@@ -0,0 +1,24 @@
+<% provide :title, t(:new_user) %>
+<h1>New account</h1>
+
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for [:supervisor, @user] do |f| %>
```
Comment (dieunb) [inline app/views/supervisor/users/new.html.erb]: user new and edit form are duplicated code

**Comment by chuanhtuan** [comment]
OK

**Pull request merged**